app/jobs/payout/gemini_job.rb

```diff
@@ -0,0 +1,23 @@
+module Payout
+  class GeminiJob < ApplicationJob
+    queue_as :scheduler
+
+    def perform(should_send_notifications: false, payout_report_id: nil)
+      publishers = Publisher.joins(:gemini_connection).with_verified_channel
+
+      publishers.find_each do |publisher|
+        IncludePublisherInPayoutReportJob.perform_async(
+          payout_report_id: payout_report_id,
+          publisher_id: publisher.id,
+          should_send_notifications: should_send_notifications,
+          kind: IncludePublisherInPayoutReportJob::GEMINI
+        )
+      end
+
+      if payout_report_id.present?
+        payout_report = PayoutReport.find(payout_report_id)
+        payout_report.update(expected_num_payments: payout_report.expected_num_payments + PayoutReport.expected_num_payments(publishers))
```

Comment:
I think this will need a lock as multiple jobs can update this row

Reply:
https://github.com/brave-intl/publishers/pull/2859/commits/9655b08f3b34246365de11746dadd59159d83e10